### ISSUE TITLE
Allow specifying a variables file with a cli option?

### ISSUE THREAD

**Issue by hgl** [cssnext/cssnext#72]
Since `postcss-custom-properties` accepts a `variables` option, I wonder if you are interested in allowing the cli to accept a json/yaml file path option, which will read its file content and pass the json object to `postcss-custom-properties`?

I can send a pull request if you are interested in merging.

**Comment by MoOx** [comment]
Maybe a more generic way to provides options from a config file for CLI ?

**Comment by hgl** [comment]
Then the `variables` json object should be directly embedded in the config file or only the file path?

My use case is that I need to expose some css variable values to my js scripts, and my plan is to save these values in a json file. cssnext will read it when processing css files, and my js files will require it before accessing it. That way, these values are available to both css and js.

It would be great if this json file only contains css variable values, and not other cssnext configs, because they are unless to js scripts.

If only the file path should be specified in the config file, I think we still should provide the option to CLI. The config file is then just a higher level structure.

What do you think?

**Comment by MoOx** [comment]
I don't really want to accept an option for just an option of an features. If anyone ask for this, we will end up with shitty spaghetti code.

By more generic I mean a config file that can like a .rc file

```console
$ cssnext --config-file cssnext.json
```
cssnext.json

```json
{
  "features": {
    "customProps": {
      "variables": {}
    }
  }
}
```

From js you can do 

```js
var cssVars = require("./cssnext.json").features.customProps.variables
```

If you need do that too many times, you can imagine a module

./cssvars.js

```js
module.exports = require("./cssnext.json").features.customProps.variables
```

Then 

```js
var cssVars = require("./cssvars.js")
```

**Comment by hgl** [comment]
I initially didn't want to take this approach because I would eventually bundle the js files, and the bundled file was going include the whole json file, not only the variables part.

But I think your concern about supporting too many options is valid, if `variables` is supported as an external file, why not supporting custom media queries also for example? The config file approach is very scalable. I can still write css variables and custom media queries to their own files and generate the config file accordingly.

Will send a pull request soon.

**Comment by bloodyowl** [comment]
@MoOx how about a `cssnext.config.js` that can require another module for variables?

**Comment by MoOx** [comment]
Yeah I was more thinking about that. But @hgl just want a simple file for variables. I don't see how we can make this simple & not super specific for cli interface.

I will open a new issue for a generic config file that can be used.

**Comment by hgl** [comment]
@bloodyowl In #74, I used something like `require(configPath)`, so if `configPath` points to a module file, it should also work.

**Comment by MoOx** [comment]
Closed by 224b6b6d0e4ebc1a92d11ebc3bd000257f5be055
